src/index.ts

```diff
@@ -591,22 +471,86 @@ class LayercodeClient implements ILayercodeClient {
     return this.wavRecorder.getStream();
   }
 
+  async setInputDevice(deviceId: string): Promise<void> {
+    try {
+      this.deviceId = deviceId;
+
+      // Restart recording with the new device
+      await this._restartAudioRecording();
+
+      // Reinitialize VAD with the new audio stream if VAD is enabled
+      if (this.vadManager.isVADEnabled()) {
+        console.log('Reinitializing VAD with new audio stream');
+        const newStream = this.wavRecorder.getStream();
+        this.vadManager.reinitialize(newStream);
+      }
+
+      // Set up amplitude monitoring
+      this._setupAmplitudeMonitoring(this.wavRecorder, this.options.onUserAmplitudeChange, (amp) => (this.userAudioAmplitude = amp));
```

Comment:
_restartAudioRecording already does _setupAmplitudeMonitoring. So we'll have duplicate monitoring now.